docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,408 @@
+In this example, that distance is 110mm (see diagram above), but if you have a different gripper, change the offset accordingly.
+
+Now that you have created the table and tissue box obstacles and the cup transform, create a WorldState that includes all of them:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a WorldState that includes the table and tissue box obstacles, and the cup transform
+world_state = WorldState(obstacles=[obstacles_in_frame], transforms = transforms)
+```
+
+## Define start and end poses
+
+You need to tell the robot where to pick up the cup, and where to put it down.
+
+While the Motion Service has the mathematical ability to plan complex motion from one position to another, around an obstacle, this is computationally intensive.
+The program will run much more efficiently if you add waypoints to more tightly define the motion.
+The way points in the code below are defined such that the path between each consecutive point can be linear, without intersecting the tissue box (or other obstacle).
+
+Though we are passing these waypoints in manually, the Motion Service will throw an error if we accidentally pass in a pose that would cause the arm or gripper to hit any obstacles.
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a start pose, where the cup starts between the gripper's jaws, on the table
+# Start pose has Z of 90mm to grab partway down the 120mm tall cup
+start_pose = Pose(x=320, y=240, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+start_pose_in_frame = PoseInFrame(reference_frame="world", pose=start_pose)
+
+# Create waypoints to increase efficiency of motion planning
+way1_pose = Pose(x=300, y=240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way1_pose_in_frame = PoseInFrame(reference_frame="world", pose=way1_pose)
+way2_pose = Pose(x=300, y=-240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way2_pose_in_frame = PoseInFrame(reference_frame="world", pose=way2_pose)
+
+# Create a pose where the cup will be set down
+end_pose = Pose(x=300, y=-250, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+end_pose_in_frame = PoseInFrame(reference_frame="world", pose=end_pose)
+```
+
+## Add a motion constraint
+
+To keep the cup upright as the arm moves it from one place on the table to another, create a [linear constraint](../../../services/motion/constraints/#linear-constraint).
+Then, when you tell the robot to move the cup from one upright position to another, the gripper will move linearly and the upright orientation of the cup will be maintained throughout the planned path.
+
+You could try using an [orientation constraint](../../../services/motion/constraints/#orientation-constraint) instead, which would also constrain the orientation.
+However, since this opens up many more options for potential paths, it is much more computationally intensive than the linear constraint.
+
```

Comment:
Since we just described the two possible constraint types we could use, let's clarify explicitly which one we'll be using, esp since most recently-described option is not the one we're using. Maybe:

>The code below creates a linear constraint, which ensures that the robot moves the cup along the predetermined route while avoiding the obstacles we've defined and while keeping the cup upright:

Reply:
I didn't want to make this super verbose and redundant, and I think the line-by-line comments can be more helpful since they say what each line is doing.
What are your thoughts on body text versus comments in the code?

Reply:
Definitely prefer explicit mention in-copy rather than relying on code comments to carry the teaching burden. Copy in code fences is not searchable in TypeSense (our search) or via external search engine's site crawlers (I think! Correct me if I'm wrong!).

Your code comments are great! But not everyone reads code comments, or reads them carefully enough to rely on them to convey core concepts. Certainly retain the code comments!! But a single sentence intro summarizing the goal and result of the 14 lines of code here helps put the code comments in context, presents conceptual content inline in the tutorial flow (caught by readers who skip code, and search engines alike!), and answers the "why" of why do I want to use this code: `which ensures that the robot moves the cup along the predetermined route while avoiding the obstacles we've defined and while keeping the cup upright` (or however you would like to choose to phrase such a summary!).

Certainly an additional paragraph would be too much, given that your line-by-line code comments cover the details well. But I think a single introductory sentence helps here, more than it risks undue verbosity.